### PR TITLE
Autogenerated documentation

### PR DESCRIPTION
**Pull request by inigomarquinez**
## Description

Autogenerates the documentation based on jsdoc using [typedoc](https://typedoc.org/).

Additionally, it deploys the documentation to GitHub pages.

## Related Issue

No related issue.

## Motivation and Context

Have friendly documentation for developers and users of the library.

## How Has This Been Tested?

No tests needed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: __tests__/e2e/cms/contentful.test.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { config as configDotenv } from 'dotenv';
 configDotenv();
 
+console.log('CONTENTFUL_ACCESS_TOKEN', process.env.CONTENTFUL_ACCESS_TOKEN);
```
Comment (matyasjay) [inline __tests__/e2e/cms/contentful.test.ts:4]:
```suggestion

```

Comment (inigomarquinez) [reply __tests__/e2e/cms/contentful.test.ts]: Opps thanks!!

**Review by matyasjay — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,14 +1,62 @@
...
+
+# If you don't have the contenful-cli installed globally
+npx contentful-cli login
+```
+
+To import the dummy content run the following command:
+
+```bash
+# If you have the contenful-cli installed globally
+contentful space import --content-file __tests__/exports/contentful/agnostic-cms-harmonizer_space.json --space-id <your-contentful-space-id>
```
Comment (Bounteous17) [inline README.md:48]:
```suggestion
npx contentful-cli space import --content-file __tests__/exports/contentful/agnostic-cms-harmonizer_space.json --space-id <your-contentful-space-id>
```

File: tsconfig.build.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
   "extends": "./tsconfig.json",
-  "exclude": ["node_modules", "test", "build"]
+  "exclude": ["node_modules", "__tests__", "doc", "build"]
```
Comment (Bounteous17) [inline tsconfig.build.json:3]: Should we add some pattern for ignoring unit tests too? Something like `*/**/*.test.ts`.
The coverage folder must be ignored too.

File: tsconfig.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
 
     /* Modules */
     "module": "commonjs" /* Specify what module code is generated. */,
-    "rootDir": "src" /* Specify the root folder within your source files. */,
+    // "rootDir": "src" /* Specify the root folder within your source files. */,
```
Comment (Bounteous17) [inline tsconfig.json:29]: Does we need to transpile some TS files outside `src/`?

**Review by Bounteous17 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (matyasjay) [reply README.md:48]: I this is not the same @Bounteous17. `npx` will run the script even if you have no CLI installed. If you apply the change, you need the reword the instructions as well.

File: tsconfig.json (hunk 4 above)
Comment (inigomarquinez) [reply tsconfig.json:29]: No, we don't. But I had to change it to make the doc generator work (typedoc). I read the docs od the library, but none of the configurations to exclude the patters worked.

File: README.md (hunk 2 above)
Comment (inigomarquinez) [reply README.md:48]: @Bounteous17 , take into account that I added two options. One if you have the cli installed globally (so you can use `contentful` and another one in the next line to use `npx`)

Comment (Bounteous17) [reply README.md:48]: @matyasjay @inigomarquinez didn't see that was for global installation usage ✅

File: tsconfig.build.json (hunk 3 above)
Comment (inigomarquinez) [reply tsconfig.build.json:3]: Done @Bounteous17 !

**Review by Bounteous17 — APPROVED** [review]

**Comment by inigomarquinez** [comment]
https://github.com/all-contributors please add @inigomarquinez  for doc,review

**Comment by inigomarquinez** [comment]
@all-contributors please add @inigomarquinez for doc,review

**Pull request merged**